common/src/model_deployment/deploy.py

```diff
@@ -0,0 +1,161 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Run Model deployment module."""
+import os
+import argparse
+from pathlib import Path
+import json
+from azure.ai.ml import MLClient
+from azure.ai.ml.entities import (
+    ManagedOnlineEndpoint,
+    ManagedOnlineDeployment,
+    OnlineRequestSettings,
+)
+from azure.identity import ManagedIdentityCredential
+from azureml.core import Run
+
+
+def parse_args():
+    """Return arguments."""
+    parser = argparse.ArgumentParser()
+
+    # add arguments
+    parser.add_argument(
+        "--registration_details",
+        type=str,
+        help="Text file that contains the ID of registered model to be deployed",
+    )
+    parser.add_argument(
+        "--endpoint_name",
+        type=str,
+        help="Name of the endpoint",
+    )
+    parser.add_argument(
+        "--deployment_name", type=str, help="Name of the the deployment"
+    )
+    parser.add_argument(
+        "--instance_type",
+        type=str,
+        help="Compute instance type to deploy model",
+        default="Standard_F8s_v2",
+    )
+    parser.add_argument(
+        "--instance_count",
+        type=int,
+        help="Number of compute instances to deploy model",
+        default=1,
+        choices=range(1, 20),
+    )
+    parser.add_argument(
+        "--model_deployment_details",
+        type=str,
+        help="Json file to which deployment details will be written",
+    )
+    parser.add_argument(
+        "--max_concurrent_requests_per_instance",
+        type=int,
+        default=1,
+        help="Maximum concurrent requests to be handled per instance",
+    )
+    parser.add_argument(
+        "--request_timeout_ms",
+        type=int,
+        default=5000,
+        help="Request timeout in ms. Max limit is 90000.",
+    )
+    parser.add_argument(
+        "--max_queue_wait_ms",
+        type=int,
+        default=500,
+        help="Maximum queue wait time of a request in ms",
+    )
+
+    # parse args
+    args = parser.parse_args()
+    print("args received ", args)
+    # return args
+
+    # Validating passed input values
+    if args.max_concurrent_requests_per_instance < 1:
```

Comment:
Validation for max limits?

Reply:
Need to figure out the better way to deal with maximum limit validation. Other than having static mapping,not able to figure out any better idea. As of now do not want to add that just for validation of one parameter.
May be we can discuss on this further and add this in next pr.
For other two parameters, I have added the max limit validation

Reply:
For them, validation is in parser, if you see there is one parameter choices = range(1,n).. So that will make sure that correct values has been passed or not